cmd/telliot/main.go

```diff
@@ -4,101 +4,28 @@
 package main
 
 import (
-	"bytes"
 	"context"
-	"fmt"
-	"math/big"
-	"net/http"
 	"os"
-	"os/signal"
 	"strconv"
-	"time"
 
-	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/common/hexutil"
-	"github.com/go-kit/kit/log"
-	"github.com/go-kit/kit/log/level"
-	cli "github.com/jawher/mow.cli"
+	"github.com/alecthomas/kong"
 	"github.com/pkg/errors"
-	"github.com/prometheus/client_golang/prometheus/promhttp"
 	tellorCommon "github.com/tellor-io/telliot/pkg/common"
 	"github.com/tellor-io/telliot/pkg/config"
 	"github.com/tellor-io/telliot/pkg/contracts"
 	"github.com/tellor-io/telliot/pkg/db"
-	"github.com/tellor-io/telliot/pkg/ops"
-	"github.com/tellor-io/telliot/pkg/rest"
 	"github.com/tellor-io/telliot/pkg/rpc"
-	"github.com/tellor-io/telliot/pkg/util"
 )
 
-var ctx context.Context
-var cont contracts.Tellor
-var acc rpc.Account
-var clt rpc.ETHClient
-var logLevel string
-var database db.DB
-var proxy db.DataServerProxy
-
-func ExitOnError(err error, operation string) {
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "%s failed: %s\n", operation, err.Error())
-		cli.Exit(-1)
-	}
-}
-
-func setup() error {
-	cfg := config.GetConfig()
-
-	err := util.SetupLoggingConfig(cfg.Logger)
-	if err != nil {
-		return errors.Wrapf(err, "parsing log config")
-	}
-
-	logLevel = cfg.LogLevel
-
-	if !cfg.EnablePoolWorker {
-		// Create an rpc client
-		client, err := rpc.NewClient(os.Getenv(config.NodeURLEnvName))
-		if err != nil {
-			return errors.Wrap(err, "create rpc client instance")
-		}
-
-		contract, err := contracts.NewTellor(cfg, client)
-		if err != nil {
-			return errors.Wrap(err, "creating contract")
-		}
-
-		account, err := rpc.NewAccount(cfg)
-		if err != nil {
-			return errors.Wrap(err, "creating account")
-		}
-		ctx = context.Background()
-		// Issue #55, halt if client is still syncing with Ethereum network
-		s, err := client.IsSyncing(ctx)
-		if err != nil {
-			return errors.Wrap(err, "determining if Ethereum client is syncing")
-		}
-		if s {
-			return errors.New("ethereum node is still syncing with the network")
-		}
-
-		clt = client
-		acc = account
-		cont = contract
-	}
-	return nil
-}
-
 // migrateAndOpenDB migrates the tx costs and deletes the db.
 // The DB is always deleted because the price avarages calculations
 // is not calculated properly between restarts.
 // TODO don't do this and just improve the price calculations.
-func migrateAndOpenDB() (db.DB, error) {
-	cfg := config.GetConfig()
+func migrateAndOpenDB(cfg *config.Config) (db.DB, db.DataServerProxy, error) {
```

Comment:
We should keep functions to a single purpose. this this change the function no longer just migrates the db, but also creates a proxy. I don't see any reason why we can't just have another function `createDbProxy`. Long term having functions with small purpose allows to make components more composable, easyer to maintain, test and refactor.

Reply:
Sounds good!